### PR TITLE
Add `Bytes` for a `[]byte` binding

### PR DESCRIPTION
**Pull request by andydotxyz**
Just adding `[]byte` to the types generated

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Review by Jacalz — COMMENTED** [review]
Should this use atomics like the other bindings, to be more consistent?

File: data/binding/comparator_helper.go [hunk 1]
```diff
@@ -11,3 +15,7 @@ func compareURI(v1, v2 fyne.URI) bool {
 	}
 	return v1.String() == v2.String()
 }
+
+func compareBytes(v1, v2 []byte) bool {
+	return bytes.Equal(v1, v2)
+}
```
Comment (Jacalz) [inline data/binding/comparator_helper.go:21]: I saw why this was added and then fixed thanks to Staticcheck. However, do we really need this wrapper function after that?

**Review by Jacalz — CHANGES_REQUESTED** [review]

File: data/binding/comparator_helper.go (hunk 1 above)
Comment (andydotxyz) [reply data/binding/comparator_helper.go:21]: Good catch, thanks

**Comment by andydotxyz** [comment]
> Should this use atomics like the other bindings, to be more consistent?

Not sure I follow here. This change just adds a new type to the list we generate. Where should I have added atomics?

**Review by stuartmscott — COMMENTED** [review]
The original suggestion was `binding.Byte` for `byte` and `binding.ByteList` for `byte[]`, what is the advantage of treating `byte[]` as a single item?

**Comment by andydotxyz** [comment]
> The original suggestion was `binding.Byte` for `byte` and `binding.ByteList` for `byte[]`, what is the advantage of treating `byte[]` as a single item?

Basically a slice of bytes is typically a single item. Like a string or an image.
The `List` bindings are more synonymous with list widgets I would think - displaying many items to a user.
I agree that semantically a byte slice is a list, but as we are focused on how things work for a GUI space it seemed reasonable.
I think it's similar to how we have `String` instead of `RuneList`.

**Comment by Jacalz** [comment]
> > Should this use atomics like the other bindings, to be more consistent?
> 
> 
> 
> Not sure I follow here. This change just adds a new type to the list we generate. Where should I have added atomics?

I thought we switched over a lot of the other binding types to use atomics instead of mutexes?

**Comment by andydotxyz** [comment]
> I thought we switched over a lot of the other binding types to use atomics instead of mutexes?

The types are all created from the same generator. Not sure what code I committed that you're saying are incorrect...

**Comment by Jacalz** [comment]
> > I thought we switched over a lot of the other binding types to use atomics instead of mutexes?
> 
> The types are all created from the same generator. Not sure what code I committed that you're saying are incorrect...

Ah, sorry. I was wrong. I thought the atom is were added to more binding types but I see now that it wasn’t the case. Ignore my concern.

File: data/binding/comparator_helper.go [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package binding
 
-import "fyne.io/fyne/v2"
+import (
+	"fyne.io/fyne/v2"
+)
```
Comment (Jacalz) [inline data/binding/comparator_helper.go:5]: This seems like an unnecessary change. I think it is a leftover from the wrapper that was removed.

**Review by Jacalz — CHANGES_REQUESTED** [review]
Just one minor nitpick. Seems sensible to me otherwise. Nice work. Seems very useful.

File: data/binding/comparator_helper.go (hunk 2 above)
Comment (andydotxyz) [reply data/binding/comparator_helper.go]: Oh yes sorry about that.

**Review by Jacalz — APPROVED** [review]
Looks good. Thanks :)
